bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1033,6 +1043,32 @@ private void setThingStatus(Thing thing, ThingStatusInfo thingStatusInfo) {
         }
     }
 
+    @Override
+    public void setDisabled(Thing thing, boolean disabled) {
+        ThingHandler handler = thing.getHandler();
+
+        if (disabled) {
+            logger.debug("Thing {} will be disabled.", thing.getUID().getAsString());
+            if (handler != null) {
+                handler.dispose();
+            }
+            setThingStatus(thing, buildStatusInfo(ThingStatus.UNINITIALIZED, ThingStatusDetail.DISABLED));
+        } else {
+            handler.initialize();
+        }
+
+        // Persist the status of the updated thing.
+        managedThingStatusInfoProvider.update(new ManagedThingStatusInfo(thing.getUID(), thing.getStatusInfo()));
+    }
+
+    private boolean isThingDisabled(Thing thing) {
+        ManagedThingStatusInfo persistedStatus = managedThingStatusInfoProvider.get(thing.getUID());
```

Comment:
Would it make sense to remove the status once a thing has been "really" removed from the system?

Reply:
How can you tell?

Reply:
To be honest ... I dont know ... The Problem I see is that a new created thing that was already previously part of the system would be disabled after re-paring by default